VerifyReplication/storage/s3.js

```diff
@@ -98,31 +87,29 @@ function getObjMd(params, cb) {
         return cb(new Error(errMsg));
     }
 
-    return client.headObject({
-        Bucket: bucket,
-        Key: key,
-        VersionId: versionId,
-    }, (err, data) => {
-        if (err) {
-            return cb(err);
+    try {
+        const commandParams = {
+            Bucket: bucket,
+            Key: key,
+        };
+        if (versionId) {
+            commandParams.VersionId = versionId;
         }
-        const {
-            ContentLength,
-            LastModified,
-            VersionId,
-            Metadata,
-        } = data;
+        
+        const data = await client.send(new HeadObjectCommand(commandParams));
         const resp = {
-            size: ContentLength,
-            lastModified: LastModified,
-            versionId: VersionId,
-            md: Metadata,
+            size: data.ContentLength,
+            lastModified: data.LastModified,
+            versionId: data.VersionId,
+            md: data.Metadata,
         };
         return cb(null, resp);
-    });
+    } catch (err) {
```

Comment:
catch must not wrap the callback: otherwise, if the callback throws, we will call it again....